suite-common/wallet-utils/src/solanaUtils.ts

```diff
@@ -50,15 +29,55 @@ export const dummyPriorityFeesForFeeEstimation: PriorityFees = {
     computeUnitLimit: '200000',
 };
 
-const addPriorityFees = async (transaction: Transaction, priorityFees: PriorityFees) => {
-    const { ComputeBudgetProgram } = await loadSolanaLib();
-    transaction.add(
-        ComputeBudgetProgram.setComputeUnitLimit({
-            units: parseInt(priorityFees.computeUnitLimit, 10),
-        }),
-        ComputeBudgetProgram.setComputeUnitPrice({
-            microLamports: parseInt(priorityFees.computeUnitPrice, 10),
-        }),
+async function createTransactionShim(message: CompilableTransactionMessage) {
+    const {
+        compileTransaction,
+        signTransaction,
+        createKeyPairFromPrivateKeyBytes,
+        getBase16Codec,
+        getTransactionEncoder,
+    } = await loadSolanaLib();
+
+    let transaction = compileTransaction(message);
+
+    return {
+        async addSignature(privateKeyHex: string) {
+            const privateKeyBytes = getBase16Codec().encode(privateKeyHex);
+            const keypair = await createKeyPairFromPrivateKeyBytes(privateKeyBytes);
+            transaction = await signTransaction([keypair], transaction);
+        },
+        serializeMessage() {
+            return getBase16Codec().decode(transaction.messageBytes);
+        },
+        serialize() {
+            return pipe(transaction, getTransactionEncoder().encode, getBase16Codec().decode);
+        },
+    };
+}
```

Comment:
Rather than to really thrash `wallet-core` I decided to make a shim that behaves mostly like the legacy API.